packages/api/src/models/animalMovementTaskModel.js

```diff
@@ -0,0 +1,103 @@
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+import Model from './baseFormatModel.js';
+import baseModel from './baseModel.js';
+import TaskModel from './taskModel.js';
+import AnimalModel from './animalModel.js';
+import AnimalBatchModel from './animalBatchModel.js';
+import AnimalMovementPurposeModel from './animalMovementPurposeModel.js';
+import AnimalMovementTaskPurposeRelationshipModel from './animalMovementTaskPurposeRelationshipModel.js';
+import AnimalMovementTaskAnimalRelationshipModel from './animalMovementTaskAnimalRelationshipModel.js';
+import AnimalMovementTaskAnimalBatchRelationshipModel from './animalMovementTaskAnimalBatchRelationshipModel.js';
+
+class AnimalMovementTask extends baseModel {
+  static get tableName() {
+    return 'animal_movement_task';
+  }
+
+  static get idColumn() {
+    return 'task_id';
+  }
+
+  // Optional JSON schema. This is not the database schema! Nothing is generated
+  // based on this. This is only used for validation. Whenever a model instance
+  // is created it is checked against this schema. http://json-schema.org/.
+  static get jsonSchema() {
+    return {
+      type: 'object',
+      required: [],
+      properties: {
+        task_id: { type: 'integer' },
+      },
+      additionalProperties: false,
+    };
+  }
+
+  static get relationMappings() {
+    return {
+      task: {
+        relation: Model.BelongsToOneRelation,
+        modelClass: TaskModel,
+        join: {
+          from: 'animal_movement_task.task_id',
+          to: 'task.task_id',
+        },
+      },
+      animals: {
+        relation: Model.ManyToManyRelation,
+        modelClass: AnimalModel,
+        join: {
+          from: 'animal_movement_task.task_id',
+          through: {
+            modelClass: AnimalMovementTaskAnimalRelationshipModel,
+            from: 'animal_movement_task_animal_relationship.task_id',
+            to: 'animal_movement_task_animal_relationship.animal_id',
+          },
+          to: 'animal.id',
+        },
+      },
+      animal_batches: {
+        relation: Model.ManyToManyRelation,
+        modelClass: AnimalBatchModel,
+        join: {
+          from: 'animal_movement_task.task_id',
+          through: {
+            modelClass: AnimalMovementTaskAnimalBatchRelationshipModel,
+            from: 'animal_movement_task_animal_batch_relationship.task_id',
+            to: 'animal_movement_task_animal_batch_relationship.animal_batch_id',
+          },
+          to: 'animal_batch.id',
+        },
+      },
+      purposes: {
```

Comment:
@SayakaOno Regarding your note on relation mappings in case you wanted feedback/discussion. 

If it helps.. I think my understanding of `relationMappings` is that you only need to specify it when you expect to use relation methods like query.relate(), graph insert, upsert etc. 

In the past, I have only thought of it as the upsert path. No upsert needed(or other relation method), no path (relationMapping) needed.

For your use case here, previously we do not intend on using a `POST: /api/animal_movement_task` to update `animal_movement_purpose` so we wouldn't add a relation mapping, and nesting shared enum table inside a task object would also be not helpful. 

But if we follow through on Antos idea to replacing the `purpose_id ` with `purpose_key` on GET endpoint then what you have here should work correctly! The consequence of this I think will be is that we may need to add something like `{unrelate: ['purposes']}` to task upsert options.

Reply:
Thank you Duncan for your comment, it's much clearer now.

I used ChatGPT's help on this (as always 😅), and the code is based on two key points:
- Joining the relations tables is not correct.
- Adding `relationMappings` is typically beneficial. It can make relationships much clearer for other developers. (However, I also learned from the Objection documentation that relation tables don’t even require a model!)

I was wondering when to define `relationMappings`, so your input is definitely helpful! If I understand correctly, adding `relationMappings` may sometimes require extra work, such as including `{unrelate: ['purposes']}`...

Replacing the `purpose_id`  with `purpose_key` was also something I wanted to discuss in tech daily.

This is a great opportunity for me to deepen my understanding of relationships, and I appreciate your insights! Were there any resources that helped you become familiar with Objection? I'd like to come back with more polished questions.

Reply:
I mostly have just been getting familiar indirectly due to the different places we use graph + relations and reading objection documentation. (I only use chatGPT for simple things like simplify this function or make jsdoc-- but it still fails me on that lol -- it will be a while before it takes my job I think 😉 )

- task + management plan use withGraphFetched to explicitly specify individual relations.
- upsertGraph() when specified with a model, I think will try to update all related (so that means I am curious whether it would be possible to update the enum table because of the `through` mechanism. If so then maybe an option like unrelate is necessary).
- graph option like `noInsert`, `noUpdate`

> Joining the relations tables is not correct.

I would be curious to learn more about this point. Our LiteFarm definition of a "relations" table may not be the same as Objections definition.  In the example with actors, movies, and movies_actors -- both actors and movies seem like what I loosely name "modules". They are "important". The enum table will only be used in one context so I do not know if that is the same kind of relations table.